components/game/context.tsx

```diff
@@ -65,13 +66,14 @@ export const GameContextProvider = ({
   }, [id]);
 
   const move = useCallback(
-    (coords: Coords) => {
+    (coords: Coords, currentPlayer: Player, gameType: GameType) => {
       setIsMoving(true);
       setMoveError(undefined);
+      const playerTurn = gameType === "automatic" ? currentPlayer : player;
       fetch("/api/game/move", {
         body: JSON.stringify({
           coords,
-          player,
+          player: playerTurn,
```

Comment:
^ to the above, which would let you to just send a `player` as it was before; it won't violate the logic: this is "the player who currently has the game opened" whether it's a bot game or a PVP game